File: 2025-1/sesion-de-clase-14/SoftProgNegocio/BOImpl/EmpleadoBOImpl.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using PUCP.Edu.Pe.SoftProg.Modelo.RRHH;
using PUCP.Edu.Pe.SoftProg.Negocio.BO;
using PUCP.Edu.Pe.SoftProg.Persistencia.DAO.RRHH;
using PUCP.Edu.Pe.SoftProg.Persistencia.DAOImpl.RRHH;

namespace PUCP.Edu.Pe.SoftProg.Negocio.BOImpl {
    public class EmpleadoBOImpl : IEmpleadoBO {
        private readonly IEmpleadoDAO empleadoDAO;

        public EmpleadoBOImpl() { 
            this.empleadoDAO = new EmpleadoDAOImpl();
        }

        public void Eliminar(int id) {
            this.empleadoDAO.Eliminar(id);
        }

        public void Guardar(Empleado empleado, Estado estado) {
            if (estado == Estado.Nuevo) {
                this.empleadoDAO.Insertar(empleado);
            }
            else if (estado == Estado.Modificar) {
                this.empleadoDAO.Modificar(empleado);
            }
        }

        public List<Empleado> Listar() {
            return this.empleadoDAO.Listar();
        }

        public Empleado Obtener(int id) {
            return this.empleadoDAO.Buscar(id);
        }
    }
}

File: 2025-1/sesion-de-clase-20/SoftProg/SoftProgNegocio/src/main/java/pe/edu/pucp/inf30/softprog/boimpl/EmpleadoBOImpl.java
package pe.edu.pucp.inf30.softprog.boimpl;

import java.util.List;
import pe.edu.pucp.inf30.softprog.bo.IEmpleadoBO;
import pe.edu.pucp.inf30.softprog.dao.rrhh.IEmpleadoDAO;
import pe.edu.pucp.inf30.softprog.daoimpl.rrhh.EmpleadoDAOImpl;
import pe.edu.pucp.inf30.softprog.model.Estado;
import pe.edu.pucp.inf30.softprog.model.rrhh.Empleado;

/**
 *
 * @author eric
 */
public class EmpleadoBOImpl implements IEmpleadoBO {
    private final IEmpleadoDAO empleadoDao;
    
    public EmpleadoBOImpl() {
        this.empleadoDao = new EmpleadoDAOImpl();
    }
    
    @Override
    public List<Empleado> listar() {
        return this.empleadoDao.listar();
    }

    @Override
    public Empleado obtener(int id) {
        return this.empleadoDao.buscar(id);
    }

    @Override
    public void eliminar(int id) {
        this.empleadoDao.eliminar(id);
    }

    @Override
    public void guardar(Empleado modelo, Estado estado) {
        if (estado == Estado.Nuevo) {
            this.empleadoDao.insertar(modelo);
        }
        else {
            this.empleadoDao.modificar(modelo);
        }
    }

    @Override
    public Empleado buscarPorDni(String dni) {
        return this.empleadoDao.buscarPorDni(dni);
    }
}
